### DISCUSSION TITLE
Why is the return type of TransportInterface::send() nullable?

### DISCUSSION THREAD

**Discussion by cheyder** [symfony/symfony#48512] [Q&A] [answered]
symfony/mailer

Does anyone know, why the [`TransportInterface::send()`](https://github.com/symfony/mailer/blob/7b355fca167fa5302c77bccdfa0af4d7abc6bd8c/Transport/TransportInterface.php#L32) returns `?SentMessage`? I couldn't find any implementation which would return null. If this could be strictly `SentMessage` it would save us some checks afterwards. Otherwise, if it is supposed to be nullable, does someone know what it is supposed to mean? Was the sending succesful, or occured any error?

**Comment by MatTheCat** [comment, marked as answer]
Before https://github.com/symfony/symfony/pull/36467 `send` would return `null` if no recipient were set but, it now seems this can indeed no longer happen. You can try opening a PR but I’m not sure it’ll be considered worth it.

**Reply by cheyder** [reply]
Thanks! That's what I was looking for. I'll prepare a pr within the next days. Let's see if it'll considered worth it ;)
